src/Harald.Infrastructure.Slack/Model/SlackChannelName.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Harald.Infrastructure.Slack.Model
+{
+    public class SlackChannelName : IValidatableObject
+    {
+        private readonly string _value;
+
+        public SlackChannelName(string value)
+        {
+            _value = value;
+        }
+
+        public override string ToString()
+        {
+            return _value;
+        }
+
+        public static implicit operator SlackChannelName(string input)
+        {
+            return new SlackChannelName(input);
+        }
+
+        public static implicit operator string(SlackChannelName input)
+        {
+            return input.ToString();
+        }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            var results = new List<ValidationResult>();
+
+            if (_value.Length > 21)
+            {
+                results.Add(new ValidationResult(string.Format(ValidationMessages.Harald_Infrastructure_Slack_Model_SlackChannelName_Validate_Length, _value, _value.Length)));
+            }
+
+            if (_value.Contains("_"))
+            {
+                results.Add(new ValidationResult(string.Format(ValidationMessages.Harald_Infrastructure_Slack_Model_SlackChannelName_Validate_Validate_Name, _value)));
+            }
+
+            return new List<ValidationResult>();
+        }
+
+        public override bool Equals(object obj)
```

Comment:
I simply copied the logic into the class from StringSubstitutable because I want to keep the domain out of the Slack facade assembly so its possible for us to package it up as a NuGet package and share it with others.